tests/scripts/all.sh

```diff
@@ -2468,6 +2468,103 @@ component_test_psa_crypto_config_accel_ecc () {
+    scripts/config.py unset MBEDTLS_ECDSA_C
+    scripts/config.py unset MBEDTLS_ECDH_C
+    scripts/config.py unset MBEDTLS_ECJPAKE_C
+
+    # Ensure also RSA_C is disabled so that the size of the public/private
+    # keys cannot be taken from there
+    scripts/config.py unset MBEDTLS_RSA_C
+    # disable key exchanges dependencies on it
+    scripts/config.py unset MBEDTLS_KEY_EXCHANGE_RSA_PSK_ENABLED
+    scripts/config.py unset MBEDTLS_KEY_EXCHANGE_RSA_ENABLED
+    scripts/config.py unset MBEDTLS_KEY_EXCHANGE_DHE_RSA_ENABLED
+    scripts/config.py unset MBEDTLS_KEY_EXCHANGE_ECDHE_RSA_ENABLED
+    scripts/config.py unset MBEDTLS_KEY_EXCHANGE_ECDH_RSA_ENABLED
+
+    # Explicitly disable all SW implementation for elliptic curves and enable
+    # their accelerated version (this excludes the built-in automatic inclusion
+    # from "config_psa.h")
+    # Just leave SW implementation for the curve with the smallest bit size
+    # (MBEDTLS_ECP_DP_SECP192R1_ENABLED) for allowing to build with ECP_C.
+    # scripts/config.py unset MBEDTLS_ECP_DP_SECP192R1_ENABLED
+    scripts/config.py unset MBEDTLS_ECP_DP_SECP224R1_ENABLED
+    scripts/config.py unset MBEDTLS_ECP_DP_SECP256R1_ENABLED
+    scripts/config.py unset MBEDTLS_ECP_DP_SECP384R1_ENABLED
+    scripts/config.py unset MBEDTLS_ECP_DP_SECP521R1_ENABLED
+    scripts/config.py unset MBEDTLS_ECP_DP_SECP192K1_ENABLED
+    scripts/config.py unset MBEDTLS_ECP_DP_SECP224K1_ENABLED
+    scripts/config.py unset MBEDTLS_ECP_DP_SECP256K1_ENABLED
+    scripts/config.py unset MBEDTLS_ECP_DP_BP256R1_ENABLED
+    scripts/config.py unset MBEDTLS_ECP_DP_BP384R1_ENABLED
+    scripts/config.py unset MBEDTLS_ECP_DP_BP512R1_ENABLED
+    scripts/config.py unset MBEDTLS_ECP_DP_CURVE25519_ENABLED
+    scripts/config.py unset MBEDTLS_ECP_DP_CURVE448_ENABLED
+    loc_accel_list="$loc_accel_list ECC_SECP_R1_521"
+    loc_accel_list="$loc_accel_list ECC_BRAINPOOL_P_R1_512"
+    loc_accel_list="$loc_accel_list ECC_MONTGOMERY_448"
+    loc_accel_list="$loc_accel_list ECC_SECP_R1_384"
+    loc_accel_list="$loc_accel_list ECC_BRAINPOOL_P_R1_384"
+    loc_accel_list="$loc_accel_list ECC_SECP_R1_256"
+    loc_accel_list="$loc_accel_list ECC_SECP_K1_256"
+    loc_accel_list="$loc_accel_list ECC_BRAINPOOL_P_R1_256"
+    loc_accel_list="$loc_accel_list ECC_MONTGOMERY_255"
+    loc_accel_list="$loc_accel_list ECC_SECP_R1_224"
+    loc_accel_list="$loc_accel_list ECC_SECP_K1_224"
+    loc_accel_list="$loc_accel_list ECC_SECP_R1_192"
+    loc_accel_list="$loc_accel_list ECC_SECP_K1_192"
+
+    # build and link with test drivers
+    loc_accel_flags="$loc_accel_flags $( echo "$loc_accel_list" | sed 's/[^ ]* */-DMBEDTLS_PSA_ACCEL_&/g' )"
+    make CFLAGS="$ASAN_CFLAGS -I../tests/include -I../tests -I../../tests -DPSA_CRYPTO_DRIVER_TEST -DMBEDTLS_TEST_LIBTESTDRIVER1 $loc_accel_flags" LDFLAGS="-ltestdriver1 $ASAN_CFLAGS"
+
+    # make sure excluded modules were not auto-re-enabled by accident
```

Comment:
We could probably go a bit further here and add something like
```
# The only built-in curve is Short Weierstrass, so ECP shouldn't have support for Montgomery curves.
# Functions with mxz in their name are specific to Montgomery curves.
not grep mxz library/ecp.o
```
This is a bit weak in that all functions with mxz in their name are static so there's no guarantee they will be visible in the `.o` file, but I tried and one of them seems to be visible at `-O2` with GCC (but not Clang). Anyway, I don't think adding that hurts, as the worst that can happen is it tests nothing.

If/when we do another component with Curve25519 as the only curve, the similar test will be more robust, as we have public function that's only defined when there are Short Weierstrass curves: `mbedtls_ecp_muladd` so we can grep for that and be certain that the result means something.

Reply:
Update = Indeed I just found 1 test that was depending on `MBEDTLS_ECP_DP_SECP192R1_ENABLED` and that didn't build with this new configuration. So it was worth to add the new test case.

Reply:
> I'm not convinced that is better to add the test for Curve25519 now

Typo for "I'm now convinced"? Anyway, I see you added it and I agree it's a good idea, thanks!

Reply:
Ops, sorry, I wrote that comment in a rush and that's the result :D